### ISSUE TITLE
Personal email not removed on deletion

### ISSUE THREAD

**Issue by AlexHenderson** [SolidOS/solidos#156]
Hello, 

I created an additional (personal) email, then decided to delete it. It no longer appears in my profile, but anonymous users can see it in the XML view. 

Alex

**Comment by jeff-zucker** [comment]
@AlexHenderson : a) what server? b) how did you delete it?

**Comment by AlexHenderson** [comment]
Hi @jeff-zucker, thanks for getting back to me so quickly (and on a holiday weekend!) 
I created an account on solidcommunity.net (https://alexhenderson.solidcommunity.net/profile/card#me). In the profile webpage I added an additional email address. Then, still using the web interface, I deleted it. 
If you look at my profile - https://alexhenderson.solidcommunity.net/profile/card#me - and select the RDF-XML view, there are two email addresses as blank nodes, but only one vcard:hasEmail entry.

**Comment by jeff-zucker** [comment]
It looks to me like what happened is that you deleted the triple asserting you have an email but not the blanknode triple asserting the email address' value.  I just tried and got the same behavior.  So it's a bug in the profile form.  Thanks for reporting, we'll add it to the task list.

**Comment by timbl** [comment]
Yes there is I suspect a general problem in the form system of not not cleaning up the twigs when you cut off a branch.
